site/app/templates/polls/NewPollPage.twig

```diff
@@ -398,6 +398,16 @@
     }
 
     $(document).ready(function() {
+        {% if poll is not defined %}
+            const today = new Date();
+            const year = today.getFullYear().toString().padStart(4, '0');
+            const month = (today.getMonth() + 1).toString().padStart(2, '0');
+            const day = today.getDate().toString().padStart(2, '0');
+            const date_string = `${year}-${month}-${day}`;
+            $("#poll-date").flatpickr({
+                defaultDate: date_string
+            });
```

Comment:
Ideally we'd be able to combine this flatpickr call with https://github.com/Submitty/Submitty/blob/96750b883980eeb41bb69fc179856012fe755e62/site/app/templates/polls/NewPollPage.twig#L188-L194 to avoid destroying and then recreating the flatpickr instance on the element, but this is probably simpler for now.